tensorrt_llm/_torch/pyexecutor/resource_manager.py

```
@@ -2894,6 +2894,47 @@ def try_allocate_generation(self, req: LlmRequest) -> bool:
+        """
+        kv_cache = self.kv_cache_map.get(req.py_request_id)
+        if kv_cache is None or not kv_cache.is_active:
+            return True
+        if history_length <= kv_cache.history_length:
+            return True
+        # resize() requires capacity >= history_length; clamp for safety.
+        target_capacity = max(kv_cache.capacity, history_length)
+        try:
+            return kv_cache.resize(target_capacity,
+                                   history_length=history_length)
+        except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)